WordPress/gutenberg
# Use the Query title block to display a "Blog" title on any template

## What?

This PR introduces a new variation to the Query Title block, enabling it to automatically display a "Blog" title on relevant templates (e.g., index and home templates) when the archive title is not available. This aims to eliminate the need for manually adding a heading block above the query loop on these templates.

## Why?

The issue was that on certain templates (like index and home), the archive title did not show anything by default. This required site creators and theme developers to manually add a heading block, which felt redundant. By adding this functionality to the Query Title block, we simplify the process and ensure that a default "Blog" title is automatically displayed when applicable.

Reference: Issue [#64567](https://github.com/WordPress/gutenberg/issues/64567)

## How?

- A new "blog title" variation is added to the Query Title block.
- The block automatically detects if it is being used on templates like index or home and displays the "Blog" title when no other archive title is available.
- This ensures the block works out of the box on these templates without additional configuration.

## Testing Instructions

1. Open a template that uses the Query Title block, such as an archive, index, or home template.
2. On the index or home templates, verify that the block now displays a "Blog" title automatically.
3. On archive or search result templates, verify that the block displays the appropriate archive title (e.g., category name, search term).
4. Verify that you no longer need to manually add a heading block above the query loop on the index or home templates.

### Testing Instructions for Keyboard

1. Navigate to the Query Title block using keyboard controls.
2. Ensure that the title is properly focused and can be edited as expected.
3. Confirm that the block works correctly when using keyboard navigation on index and home templates (displays "Blog") and on archive templates (displays the correct archive title).
